crates/rolldown_debug/src/debug_formatter.rs

```diff
@@ -53,19 +57,29 @@ where
         format!(".rolldown/{}/log.json", build_id.0)
       });
 
-    let mut file = match OpenOptions::new().create(true).append(true).open(&log_filename) {
-      Ok(v) => v,
-      Err(e) => match e.kind() {
-        std::io::ErrorKind::ReadOnlyFilesystem => {
-          // WASI environment, we can't write to the filesystem
-          return Ok(());
-        }
-        _ => {
-          // Other errors, we just return the error
-          return Err(std::fmt::Error);
-        }
-      },
-    };
+    // TODO(hyf0): While this prevents memory leaks, we should come up with a better solution.
+    if OPEN_FILES.len() > 10 {
+      let oldest_file = OPEN_FILES.iter().next();
```

Comment:
Does this assumption always hold? Since it is not IndexMap

Reply:
Not really matter. It's ok we don't really remove the last one. This is only a temporary solution.

Reply:
I plan to bind these files to the `Bundler` instance and remove them when bundler get dropped.